### DISCUSSION TITLE
How do I access `VERCEL_ENV` and `VERCEL_URL` environment variables on client side in Next?

### DISCUSSION THREAD

**Discussion by paraswtf** [vercel/next.js#45802] [Help] [answered]
I want to access the `VERCEL_URL` variables and `VERCEL_ENV` variables on the client side to avoid creating additional versions of this manually with the `NEXT_PUBLIC_` prefix. Since those values will not be automatically updated when I change the domain for example.
Is there any way to achieve this?
Or possibly a feature Vercel could have? (like using other `system env variables` as variables in other `env variables` to achieve this)

**Comment by paraswtf** [comment, marked as answer]
I found the answer here:
https://vercel.com/docs/concepts/projects/environment-variables#radix-:R10mp9il8m:-trigger-tab1

**Reply by kellyrmilligan** [reply]
I don't see any answer on this page as of 5/23/2024

**Reply by kellyrmilligan** [reply]
https://vercel.com/docs/projects/environment-variables/system-environment-variables#framework-environment-variables

**Reply by pollencio** [reply]
More specifically, here: https://vercel.com/docs/projects/environment-variables/framework-environment-variables#NEXT_PUBLIC_VERCEL_ENV

**Comment by FNDEVVE** [comment]
In case anyone has the same issue as me - when you turn on "Automatically expose System Environment Variables" you need to fetch "NEXT_PUBLIC_VERCEL_ENV" instead of just "VERCEL_ENV".

**Reply by kellyrmilligan** [reply]
https://vercel.com/docs/projects/environment-variables/system-environment-variables#framework-environment-variables

**Reply by andrebautista** [reply]
Thanks @kellyrmilligan 😄

**Reply by robhadfield** [reply]
THIS should be the highlighted answer... why is there no mention of this in the docs? Thanks!

**Reply by nodkrot** [reply]
"Automatically expose System Environment Variables" is not there

**Reply by tigeryy2** [reply]
https://vercel.com/docs/environment-variables/framework-environment-variables

The other link wasn't working for me...

**Reply by 50bbx** [reply]
What about the [VERCEL](https://vercel.com/docs/environment-variables/system-environment-variables#VERCEL) env variable? There is no `NEXT_PUBLIC_VERCEL` equivalent. Why?
